### PR TITLE
Added support for new backend system (Azure DevOps, DevTools, and Linguist)

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

Added alpha support for the new backend system for the Azure DevOps, DevTools, and Linguist backend plugins 🚀 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Nice!

Hey I know you're aware of #18301, and I think the intent there is that backend plugins now should not be alpha exports, but that they can be regular public ones instead!

File: plugins/devtools-backend/README.md [hunk 1]
```diff
@@ -49,6 +49,34 @@ Here's how to get the DevTools Backend up and running:
...
+
+```diff
+  import { createBackend } from '@backstage/backend-defaults';
+  import { appPlugin } from '@backstage/plugin-app-backend';
+  import { catalogPlugin } from '@backstage/plugin-catalog-backend';
+  import {
+    scaffolderPlugin,
+    catalogModuleTemplateKind,
+  } from '@backstage/plugin-scaffolder-backend';
++ import { devtoolsPlugin } from '@backstage/plugin-devtools-backend/alpha';
```
Comment (freben) [inline plugins/devtools-backend/README.md:66]: In all these instructions, would it make sense to make them even more compact down to the essentials? I know some people are helped by extra context but some are also confused by it ("my file doesn't have the appPlugin line, is that necessary?"). Could perhaps shorten to something like

```ts
import { devtoolsPlugin } from '@backstage/plugin-devtools-backend';

const backend = createBackend();
backend.add(devtoolsPlugin());
// ... other feature additions

backend.start();
```

or so?

**Comment by awanlin** [comment]
> Nice!
> 
> Hey I know you're aware of #18301, and I think the intent there is that backend plugins now should not be alpha exports, but that they can be regular public ones instead!

Sounds good to me! When I was looking at examples in the code base it was a bit of both so I wasn't sure, figured I would go the alpha route and if that wasn't right someone would tell me so thanks for that :)

File: plugins/devtools-backend/README.md (hunk 1 above)
Comment (awanlin) [reply plugins/devtools-backend/README.md:60]: Makes sense to me, done 👍

**Comment by awanlin** [comment]
I've made all the requested changes, seeing as @freben is on vacation @benjdlambert this falls to you but I'm fine if this doesn't happen right away 👍

File: plugins/linguist-backend/package.json [hunk 2]
```diff
@@ -9,6 +9,17 @@
     "main": "dist/index.cjs.js",
     "types": "dist/index.d.ts"
   },
+  "exports": {
```
Comment (benjdlambert) [inline plugins/linguist-backend/package.json:12]: Can probably clean up these changes in the package.jsons -  they're not needed right as there's no alpha exports anymore?

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Nice thanks! :tada:

**Comment by awanlin** [comment]
Thanks @benjdlambert 🚀

**Pull request merged**